lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -1066,9 +1076,20 @@ def mean_var(x, count, count_sum):
     ensemble_mean, ensemble_mean_var = mean_var(
         np.array([c.value for c in cve_based]), counts, counts_sum
     )
-    ensemble_localization_var, var_of_localization_var = mean_var(
-        np.array([c.localization_variance for c in cve_based]), counts, counts_sum
-    )
+
+    if not valid_localization_var:
+        warnings.warn(
+            RuntimeWarning(
+                "Localization variances cannot be reliably calculated for an ensemble of "
+                "tracks from kymographs with different line times or pixel sizes."
+            ),
+            stacklevel=2,
+        )
```

Comment:
To me, having this warning here, seems a bit smelly because the message contains details that you shouldn't have at this level. I would make it optional to calculate it here, but put the warning upstream.

Reply:
hah that was what I originally did, but looking at the rest of the API it seemed that the warnings/errors were triggered at the low level API so I changed it. I'll switch it back